oidc-wp/openid-connect-generic
# Adding acr_values support

### All Submissions:


### Changes proposed in this Pull Request:

This PR enhances the plugin with capabillities on handling acr_values - in which the request to the IDP includes a specific authentication contract as defined on the IDP.
Next to that enhancements have been made to also check - when set that the - if defined authentication contract is honored.


### How to test the changes in this Pull Request:

1. Define the added acr_values field option with a contract that exists on the IDP ie. urn:domain:contract:my_method_of_auth ( this is the exact name of the authentication uri as defined on a/the IDP.
2. Test login, and see that the added (acr_-)values to the url will influence the behaviour of the authentication contract with the IDP.
3. Verify manual url-manipulation check so it will error / disallow login ( remove the added acs_values from url manually and try it on your IDP)

### Other information:


### Changelog entry

> Enter a summary of all changes on this Pull Request. This will appear in the changelog if accepted.

- Added acr_values support & verification checks that it when defined in options is honored.
-
